llama_stack/providers/utils/inference/embedding_mixin.py

```diff
@@ -37,13 +38,12 @@ async def embeddings(
         task_type: Optional[EmbeddingTaskType] = None,
     ) -> EmbeddingsResponse:
         model = await self.model_store.get_model(model_id)
-        embedding_model = self._load_sentence_transformer_model(model.provider_resource_id)
-        embeddings = embedding_model.encode(
-            [interleaved_content_as_str(content) for content in contents], show_progress_bar=False
-        )
+        embedding_model = await self._load_sentence_transformer_model(model.provider_resource_id)
```

Comment:
If the model has yet to be downloaded, this could block for the duration of the download, it would need to also be downloaded in a async fashion